src/overlays/actors/ovl_Dm_An/z_dm_an.c

```diff
@@ -31,37 +33,336 @@ const ActorInit Dm_An_InitVars = {
+                this->unk_2B0 = action;
+                this->unk_2D4 = 1;
+                func_80C1C4D8(this, globalCtx, sp28[action]);
+            }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_An/func_80C1C8E8.s")
+            switch (this->unk_2B0) {
+                case 2:
+                case 4:
+                case 5:
+                case 6:
+                case 7:
+                case 8:
+                    if ((this->unk_2C8 == 12) || (this->unk_2C8 == 4) || (this->unk_2C8 == 6) || (this->unk_2C8 == 8)) {
+                        if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame) != 0) {
+                            func_80C1C4D8(this, globalCtx, this->unk_2C8 + 1);
+                        }
+                    }
+                    break;
+            }
+            Cutscene_ActorTranslateAndYaw(&this->actor, globalCtx, sp20);
+        }
+    } else if (this->unk_2D0 != 0) {
+        this->unk_2D0 = 0;
+        this->unk_2D4 = 0;
+        func_80C1C4D8(this, globalCtx, this->unk_2CC);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_An/func_80C1C958.s")
+void func_80C1CC80(DmAn* this, GlobalContext* globalCtx) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_An/func_80C1CAB0.s")
+void DmAn_Init(Actor* thisx, GlobalContext* globalCtx) {
+    DmAn* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_An/func_80C1CC80.s")
+    this->unk_2AC = SubS_GetObjectIndex(OBJECT_AN4, globalCtx);
+    this->unk_2AD = SubS_GetObjectIndex(OBJECT_MSMO, globalCtx);
+    this->actionFunc = func_80C1C958;
+}
+
+void DmAn_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void DmAn_Update(Actor* thisx, GlobalContext* globalCtx) {
+    DmAn* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+
+    func_80C1C83C(this, globalCtx);
+
+    if (this->actor.draw != NULL) {
+        func_80C1C410(this, globalCtx);
+        func_80C1C5B4(this);
+    }
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 30.0f, 12.0f, 0.0f, 4);
+}
+
+Vec3f D_80C1D2C8 = { 450.0f, 700.0f, -760.0f };
+Vec3s D_80C1D2D4 = { 0x238C, 0, -0x3FFC };
+Vec3f D_80C1D2DC = { 1000.0f, 0.0f, 0.0f };
+
+void func_80C1CD80(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    s32 pad[2];
+    DmAn* this = THIS;
+    s8 sp2B = this->actor.objBankIndex;
+    s8 sp2A = this->unk_2AD;
+
+    if ((limbIndex == OBJECT_AN1_LIMB_05) && (this->unk_2D4 != 0)) {
+        OPEN_DISPS(globalCtx->state.gfxCtx);
+
+        Matrix_StatePush();
+        Matrix_JointPosition(&D_80C1D2C8, &D_80C1D2D4);
+
+        gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPSegment(POLY_OPA_DISP++, 0x06, globalCtx->objectCtx.status[sp2A].segment);
+        gSPDisplayList(POLY_OPA_DISP++, gMoonMaskDL);
+        gSPSegment(POLY_OPA_DISP++, 0x06, globalCtx->objectCtx.status[sp2B].segment);
```

Comment:
sp2A and sp2B have to be local variables?

If they are required to match, can they be moved into the if code block?

Reply:
They're quired and no, they have to be up to.